### PR TITLE
Fix Symbol$CompletionFailure crash when building CFG

### LINKED ISSUE #6396: `Symbol$CompletionFailure` crash when building CFG
To reproduce clone the following repo:

https://github.com/msridhar/cf-cfg-crash-example

Then run `./gradlew compileJava`.  I see (with CF 3.42.0):

```
> Task :lib:compileJava FAILED
error: SourceChecker.typeProcess: unexpected Throwable (CompletionFailure) while processing /Users/msridhar/git-repos/cf-cfg-crash-example/lib/src/main/java/net/sridharan/cfgcrash/Library.java; message: class file for org.apache.spark.sql.SparkSession$1 not found
  ; The Checker Framework crashed.  Please report the crash.  Version: Checker Framework 3.42.0.
  Compilation unit: /Users/msridhar/git-repos/cf-cfg-crash-example/lib/src/main/java/net/sridharan/cfgcrash/Library.java
  Last visited tree at line 8 column 1:
  public class Library {
  Exception: com.sun.tools.javac.code.Symbol$CompletionFailure: class file for org.apache.spark.sql.SparkSession$1 not found; com.sun.tools.javac.code.Symbol$CompletionFailure: class file for org.apache.spark.sql.SparkSession$1 not found
1 error
```

I see the same crash with CF 3.41.0 but not with 3.40.0.  The class `org.apache.spark.sql.SparkSession$1` does not exist in the library `org.apache.spark:spark-sql_2.12:3.3.2`.  I couldn't convince Gradle / javac to print a full stack trace here, sorry.  We can work around this issue in NullAway by downgrading to Checker Framework 3.40.0.

When debugging the same crash in NullAway I found that it occurs during phase 1 of CFG construction for [this method](https://github.com/msridhar/cf-cfg-crash-example/blob/779c7aabfe12d89087d11f7aed7fa6b6b6fa4559/lib/src/main/java/net/sridharan/cfgcrash/Library.java#L13-L18).  I wonder if this has to do with the new try-with-resources desugaring; will take a look.

### PR DESCRIPTION
**Pull request by msridhar**
Fixes #6396 

It turns out the crash occurred inside the `javax.lang.model.util.Elements#getAllMembers` method, which was called to find the implementation of `close()` to invoke for a resource in a try-with-resources block.  I don't know the exact cause, but it looks like that method tries to resolve all members in all superclasses of the argument, which is leading to a crash with this specific third-party jar file.  I suspect the root cause of the crash is somewhere else.  (Maybe in the Scala compiler?  The jar seems to include Scala-generated classes.)  But, as a workaround, it seems that if we search for the method more directly, only looking for members with the name `"close"`, the crash is avoided.  I'm not 100% thrilled with this fix, but it's the best I can come up with at this point to avoid this crash.

### CODE REVIEW

**Review by mernst — APPROVED** [review]
Thanks!

**Comment by mernst** [comment]
@msridhar Something in this pull request significantly degrades run-time performance.  I've run CI 3 times and wasn't able to complete within the timeout any of those times.  Could you look into improving that?  Maybe use the original code, but catch the exception and if the exception is caught, use your new code.

**Comment by msridhar** [comment]
@mernst I'm puzzled by the CI failure.  It seems the only job failing is `junit_jdk17`, but `junit_jdk11` and `junit_jdk21` are both fine.  And, I can run `./gradlew :checker:test` on JDK 17 on my machine and I see no degradation in performance.  Can I somehow increase logging on CI for the `junit_jdk17` job to get a better idea of what could be happening, maybe by adding `--info` to the Gradle command?  I can try poking at it by using the old code and catching the exception if it fails, but even if that works somehow (I am skeptical it will), I'd like to understand the root cause.

**Comment by mernst** [comment]
@msridhar I didn't see anything obviously non-performant in your code.  Feel free to add any logging you want to this pull request; we can remove or comment it out before merging.  I just triggered a new build (as opposed to re-running the failed build).

**Comment by mernst** [comment]
I'm seeing the failure on the master branch of my fork (https://github.com/mernst/checker-framework/commits/master/).  I'm running some other tests now.

File: javacutil/src/main/java/org/checkerframework/javacutil/trees/TreeBuilder.java [hunk 1]
```diff
@@ -145,10 +146,21 @@ public MemberSelectTree buildCloseMethodAccess(ExpressionTree autoCloseableExpr)
...
 
-    for (ExecutableElement method : ElementFilter.methodsIn(elements.getAllMembers(exprElement))) {
-      if (method.getParameters().isEmpty() && method.getSimpleName().contentEquals("close")) {
-        closeMethod = (Symbol.MethodSymbol) method;
-        break;
+    // We could use elements.getAllMembers(exprElement) to find the close method, but in rare cases
+    // calling that method crashes with a Symbol$CompletionFailure exception.  See
+    // https://github.com/typetools/checker-framework/issues/6396.  The code below directly searches
+    // all supertypes for the method and avoids the crash.
+    Name closeName = names.fromString("close");
```
Comment (smillst) [inline javacutil/src/main/java/org/checkerframework/javacutil/trees/TreeBuilder.java:153]: I think this call is potentially costly, which is why the Checker Framework doesn't use `Name`s very often.  So you could try making it a field.

**Comment by msridhar** [comment]
@mernst @smillst could I get a re-review on this one?  No significant changes since the last approval

**Review by smillst — APPROVED** [review]

**Pull request merged**